pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/json/ImmutableJsonIndexReader.java

```diff
@@ -345,6 +370,103 @@ public Map<String, RoaringBitmap> getMatchingDocsMap(String key) {
     return matchingDocsMap;
   }
 
+  @Override
+  public Map<String, RoaringBitmap> getValueToFlattenedDocIdsMap(String jsonPathKey,
```

Comment:
Directly modify the getMatchingDocsMap which should have return FlattenDocIds and move the converting of flattenDocIds to DocIds into the up layer, i.e. transform function. I understand fixing the unit test JsonIndexTest might take more time but it's worth to do that since more test cases can be reused.

Reply:
Ack